common-lib/src/cmr/common/generics.clj

```diff
@@ -117,3 +119,118 @@
                        (get (json/parse-string index-raw true) :SubConceptType "X")))))
           {}
           (latest-approved-documents)))
+
+(defn read-generic-doc-file
+  "Return the specific schema's documentation files given the schema keyword name and version number.
+   if the file cannot be read, return an empty string which will have no impact on the API document.
+   Parameters:
+   * file-name: [ingest | search]
+   * generic-keyword: [:grid | ...]
+   * generic-version: 0.0.1
+   Returns: string"
+  [file-name generic-keyword generic-version]
+  (try
+    (-> "schemas/%s/v%s/%s.md"
+        (format (name generic-keyword) generic-version (name file-name))
+        (io/resource)
+        (slurp))
+    (catch Exception e (info (format "generic %s was skipped" generic-keyword))
+           (str ""))))
+
+(defn all-generic-docs
+  "Parse over all of the generic documents and return their combined markdown as a string
+  Parameters:
+  * file-name: [ingest | search]
+  Returns: string"
+  [file-name]
+  (string/join (seq (for [[k,v] (latest-approved-documents)]
+                      (read-generic-doc-file file-name k (str v))))))
+
+(def search-depth-map (hash-map 3 0 4 4 5 8))
```

Comment:
this really needs an explanation

Reply:
And echoing @ygliuvt concernt about common knowing about ingest and search

Reply:
(map
   (fn [x]
     (println x "=" (* 4 (- x 3))))
   [3 4 5 6])

Reply:
Using case statement passed from the app so common has no notion of search or ingest